packages/transformers/js/src/visitors/utils.js

```diff
@@ -70,6 +72,40 @@ export function isInFalsyBranch(ancestors: Array<Node>) {
   });
 }
 
+export function deleteClosestFalsyBranch(ancestors: Array<Node>) {
+  const testNode = ancestors[ancestors.length - 1];
+  let conditionNode = [...ancestors]
+    .reverse()
+    .find(node => isIfStatement(node) || isConditionalExpression(node));
+  if (conditionNode) {
+    const isClosestTestNode = conditionNode.test === testNode;
+    const isClosestUnaryExpression =
+      isUnaryExpression(conditionNode.test) &&
+      conditionNode.test.argument === testNode;
+    const isClosestBinaryExpression =
+      isBinaryExpression(conditionNode.test) &&
+      (conditionNode.test.left === testNode ||
+        conditionNode.test.right === testNode);
+
+    if (
+      isClosestTestNode ||
+      isClosestUnaryExpression ||
+      isClosestBinaryExpression
+    ) {
+      const res = evaluateExpression(conditionNode.test);
+      if (res && res.confident) {
+        const index = ancestors.indexOf(conditionNode);
+        ancestors[index] = conditionNode.alternate;
+        if (res.value) {
+          Object.assign(conditionNode, conditionNode.consequent);
```

Comment:
This was the best I came up with to mutate the node/ remove the dead branch. Are there better ways? Feels dirty to me.

Reply:
There isn't, but you should use the `morph` function defined above, maybe that makes you feel better 😄